packages/firebase_data_connect/firebase_data_connect/example/lib/main.dart

```diff
@@ -0,0 +1,270 @@
+// Copyright 2024, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:firebase_app_check/firebase_app_check.dart';
+import 'package:firebase_data_connect_example/login.dart';
+import 'package:flutter/foundation.dart';
+
+import 'firebase_options.dart';
+import 'package:flutter/material.dart';
+import 'package:firebase_data_connect/firebase_data_connect.dart';
+import 'package:firebase_core/firebase_core.dart';
+import 'package:flutter_rating_bar/flutter_rating_bar.dart';
+
+import 'generated/movies.dart';
+
+void main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+  await Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform);
+  runApp(const MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter DataConnect Demo',
+      theme: ThemeData(
+        colorScheme: ColorScheme.fromSeed(seedColor: Colors.blue),
+        useMaterial3: true,
+      ),
+      home: const Login(),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  const MyHomePage({super.key, required this.title});
+  final String title;
+
+  @override
+  State<MyHomePage> createState() => _MyHomePageState();
+}
+
+class DataConnectWidget extends StatefulWidget {
+  const DataConnectWidget({super.key});
+  @override
+  State<DataConnectWidget> createState() => _DataConnectWidgetState();
+}
+
+class _DataConnectWidgetState extends State<DataConnectWidget> {
+  final TextEditingController _genreController = TextEditingController();
+  final TextEditingController _titleController = TextEditingController();
+  DateTime _releaseYearDate = DateTime(1920);
+  List<ListMoviesMovies> _movies = [];
+  double _rating = 0;
+
+  Future<void> triggerReload() async {
+    // await appcheck();
+    QueryRef ref = MoviesConnector.instance.listMovies.ref();
+
+    ref.execute().ignore();
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    String host = 'localhost';
+    try {
+      if (Platform.isAndroid) {
+        host = '10.0.2.2';
+      }
+    } catch (_) {}
```

Comment:
For other plugins, we hide this mechanism inside the `useEmulator` method, see here:
```
  Future<void> useAuthEmulator(String host, int port,
      {bool automaticHostMapping = true}) async {
    String mappedHost = host;

    if (!kIsWeb && defaultTargetPlatform == TargetPlatform.android) {
      if ((mappedHost == 'localhost' || mappedHost == '127.0.0.1') &&
          automaticHostMapping) {
        // ignore: avoid_print
        print('Mapping Auth Emulator host "$mappedHost" to "10.0.2.2".');
        mappedHost = '10.0.2.2';
      }
    }

    await _delegate.useAuthEmulator(mappedHost, port);
  }

```

Reply:
Moved to `firebase_common` and called in `firebase_data_connect.dart`